nixos/modules/services/hardware/powerstation.nix

```diff
@@ -0,0 +1,37 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.powerstation;
+in
+{
+  options.services.powerstation = {
+    enable = lib.mkEnableOption "PowerStation";
+    package = lib.mkPackageOption pkgs "powerstation" { };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+
+    systemd.services.powerstation = {
+      description = "PowerStation Service";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "graphical-session.target" ];
+      environment = {
+        XDG_DATA_DIRS = "/run/current-system/sw/share";
+      };
+
+      serviceConfig = {
+        User = "root";
+        Group = "root";
```

Comment:
Any chance this can be run as a non-root user with capabilities?

Reply:
Not easily. PowerStation uses several sysfs kernel interfaces in order to manage the GPU TDP and CPU features.